python/lsst/sims/utils/timeTransformations.py

```diff
@@ -0,0 +1,195 @@
+"""
+For definitions of all (or, at least, most of) the time systems handled in this file, see
+
+https://www.cv.nrao.edu/~rfisher/Ephemerides/times.html
+"""
+
+import numpy as np
+import warnings
+import palpy
+from lsst.sims.utils import Ut1MinusUtcData
+from lsst.sims.utils import TT_from_TAI_Data
+
+__all__ = ["taiFromUtc", "utcFromTai",
+           "dutFromUtc",
+           "ut1FromUtc", "utcFromUt1",
+           "dttFromUtc", "ttFromTai",
+           "tdbFromTt"]
+
+
+def taiFromUtc(utc):
+    """
+    Use the palpy method palDat to convert from Coordinated Universal Time
+    (UTC) to International Atomic Time (TAI)
+
+    @param [in] utc is the UTC time as an MJD
+
+    @param [out] TAI time as an MJD
+    """
+
+    dt = palpy.dat(utc) # returns TAI-UTC in seconds
+    return utc + dt/86400.0
+
+
+def utcFromTai(tai):
+    """
+    Use the palpy method palDat to convert from International Atomic Time
+    (TAI) to Coordinated Universal Time (UTC)
+
+    @param [in] tai is the TAI time as an MJD
+
+    @param [out] utc time as an MJD
+    """
+
+    # because the PALPY method only returns TAI-UTC as a function
+    # of UTC, we will have to construct an array of TAI and UTC
+    # values using taiFromUtc and interpolate along them
+
```

Comment:
So I see that in the underlying erfa library there are more functions that Tim could be using instead of just eraDat (which gives you palpy.dat). You could ask him to add the other relevant time conversion routines to palpy and then not have to do this interpolation game. Or we could use erfa directly. I think we would still need to get the DUT tables and read the values from there to feed to the erfa (or pal) routines. 

Note that there is an eraTaiUtc and an eraUtcTai, as well as eraTaiUt1, eraTaiTT, as well as tt/tdb conversions in both directions (which would avoid your problem about the tdb not being completely accurate).